sqlitecluster/SQLite.cpp

```diff
@@ -677,6 +677,7 @@ bool SQLite::prepare(uint64_t* transactionID, string* transactionhash) {
     static const size_t deleteLimit = 10;
     if (minJournalEntry < oldestCommitToKeep) {
         auto startUS = STimeNow();
+        shared_lock<shared_mutex> lock(_sharedData.writeLock);
```

Comment:
This was missed here: https://github.com/Expensify/Bedrock/pull/2024

Without this lock, we could still do journal deletes during a ZFS snapshot. This shound fix that problem.